### ISSUE TITLE
Markdown Tables cannot merge cells both horizontally and vertically

### ISSUE THREAD

**Issue by IStr666** [doxygen/doxygen#8864]
With Markdown tables you can:
- merge cells vertically using ^
- merge cells horizontally by using ||
- but if you try to do both the finished table is wrong.

e.g.

```
| Cell1| Cell2 | Cell3 |
|------|-------|-------|
| merge right ||       |
| merge|       |       |
| below|^      |^      |
| merge both  || value1|
|^            || value2|

```

![Capture](https://user-images.githubusercontent.com/93402409/139483552-c1356cef-d2b2-4fa1-af78-aca786fbea70.PNG)

Screenshot taken from Firefox 93.0 on Windows 10

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8865

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
